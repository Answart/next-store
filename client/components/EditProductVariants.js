import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyledEditProductVariants } from './styles/ProductStyles';
import DisplayMessage from './DisplayMessage';
import Product from './Product';
import { UpdateProductVariantForm } from './Forms';
import { DeleteProductVariant } from './Buttons';


class EditProductVariants extends Component {
  static propTypes = {
    product: PropTypes.shape({
      id: PropTypes.string.isRequired,
      department: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      category: PropTypes.string,
      brand: PropTypes.string,
      online: PropTypes.bool.isRequired,
      image: PropTypes.shape({
        id: PropTypes.string.isRequired,
        cloudinary_id: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        width: PropTypes.number.isRequired,
        height: PropTypes.number.isRequired,
        transformation: PropTypes.string.isRequired,
        image_url: PropTypes.string.isRequired,
        large_image_url: PropTypes.string.isRequired
      }).isRequired,
      variants: PropTypes.array.isRequired,
    }).isRequired
  };
  state = {
    currentVariant: null,
    message: ''
  };
  selectVariant = (e, currentVariant, message = '') => {
    if (!!e && e.preventDefault) e.preventDefault();

    if (!this.props.product.variants.length) return;
    this.setState({ currentVariant: null, message }, () => {
      if (currentVariant) this.setState({ currentVariant, message });
    });
  }
  render() {
    const { currentVariant } = this.state;
    const id = currentVariant ? currentVariant.id : null;
    return (
      <StyledEditProductVariants>
        <DisplayMessage success={this.state.message} />

        {!currentVariant ? (
          <div className="edit-prdct-var-choose">
            <div className="edit-prdct-lbl">
              1. Choose Selection to Update
            </div>

            <div className="edit-prdct-content">
              <Product
                product={this.props.product}
                demoView={true}
                variantAction={this.selectVariant}
                variantActionLabel='Select'
              />
            </div>
          </div>
        ) : (
          <div className='edit-prdct-var-form'>
            <div className="edit-prdct-lbl">
              2. Update Selection
            </div>

            <UpdateProductVariantForm
              variant={currentVariant}
              goBack={(e) => this.selectVariant(e, null)}
            />

            <div className="edit-pg-content-footer">
              <DeleteProductVariant
                id={id}
                productId={this.props.product.id}
                postDelete={(e) => this.selectVariant(e, null, `Deleted selection with size '${currentVariant.size}' and color '${currentVariant.color}'.`)}
              >Delete Selection</DeleteProductVariant>
            </div>
          </div>
        )}
      </StyledEditProductVariants>
    );
  };
};

export default EditProductVariants;
