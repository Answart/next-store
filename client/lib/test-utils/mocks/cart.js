import {
  ADD_TO_CART_MUTATION,
  UPDATE_CARTITEM_MUTATION,
  REMOVE_FROM_CART_MUTATION
} from '../../../graphql'
import { fakeCartItem } from '../utils';
import { mockVariant } from './variant';

const mockCartItem = fakeCartItem();


const addToCartMutationMock = {
  request: { query: ADD_TO_CART_MUTATION, variables: { id: mockVariant.id } },
  result: {
    data: {
      addToCart: {
        __typename: mockCartItem.__typename,
        id: mockCartItem.id,
      },
    },
  },
};

const updateCartItemMutationMock = (quantity) => ({
  request: { query: UPDATE_CARTITEM_MUTATION, variables: { id: mockCartItem.id, quantity } },
  result: {
    data: {
      updateCartItem: {
        __typename: mockCartItem.__typename,
        id: mockCartItem.id,
        quantity,
        variant: { ...mockCartItem.variant }
      },
    },
  },
});

const removeFromCartMutationMock = {
  request: { query: REMOVE_FROM_CART_MUTATION, variables: { id: mockCartItem.id } },
  result: {
    data: {
      removeFromCart: {
        __typename: mockCartItem.__typename,
        id: mockCartItem.id,
      },
    },
  },
};


export {
  mockCartItem,
  addToCartMutationMock,
  updateCartItemMutationMock,
  removeFromCartMutationMock
}
