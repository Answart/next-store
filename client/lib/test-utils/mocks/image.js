import { CREATE_IMAGE_MUTATION } from '../../../graphql'
import { fakeImage } from '../utils';

const mockImage = fakeImage();
const mockImageVariables = {
  cloudinary_id: mockImage.cloudinary_id,
  name: mockImage.name,
  height: mockImage.height,
  width: mockImage.width,
  transformation: mockImage.transformation,
  image_url: mockImage.image_url,
  large_image_url: mockImage.large_image_url
}


const createImageMutationMock = (overrides) => ({
  request: {
    query: CREATE_IMAGE_MUTATION,
    variables: {
      ...mockImageVariables,
      ...overrides,
    },
  },
  result: {
    data: {
      createImage: {
        __typename: 'Image',
        id: mockImage.id,
      },
    },
  },
});

const createImageMutationErrorMock = {
  request: {
    query: CREATE_IMAGE_MUTATION,
    variables: {
      ...mockImageVariables,
      cloudinary_id: '',
      name: '',
    },
  },
  result: {
    errors: [{ message: 'ack!' }],
  }
};


export {
  mockImage, mockImageVariables,
  createImageMutationMock, createImageMutationErrorMock,
}
