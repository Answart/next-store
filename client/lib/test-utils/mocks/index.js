export {
  mockUser, mockUsers,
  mockImage, mockImageVariables,
  mockProduct, mockProducts, mockShopProductsVariables,
  mockVariant,
  mockCartItem,
  mockOrderItem, mockOrder,
} from './typeDefs';
export {
  localUserQueryEmptyCartMock, localUserQueryNoUserMock, localUserQueryCartItemMock,
  userQueryEmptyCartMock, userQueryNoUserMock, userQueryCartItemMock,
  signupMutationMock,
  signinMutationMock,
  signoutMutationMock,
  requestPasswordResetMutationMock, requestPasswordResetMutationErrorMock,
  resetPasswordMutationMock, resetPasswordMutationErrorMock,
  updatePermissionsMutationMock,
  createImageMutationMock, createImageMutationErrorMock,
  productQueryMock, productQueryErrorMock, productQueryNoProductMock, productQueryNoVariantMock,
  shopProductsQueryFilterMock,
  createProductMutationMock, createProductMutationErrorMock,
  updateProductMutationMock, updateProductMutationErrorMock,
  deleteProductMutationMock,
  createProductVariantMutationMock, createProductVariantMutationErrorMock,
  updateProductVariantMutationMock, updateProductVariantMutationErrorMock,
  deleteProductVariantMutationMock,
  addToCartMutationMock,
  updateCartItemMutationMock,
  removeFromCartMutationMock,
} from './resolvers';
