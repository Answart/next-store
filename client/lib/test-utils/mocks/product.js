import {
  PRODUCT_QUERY, SHOP_PRODUCTS_QUERY,
  CREATE_PRODUCT_MUTATION,
  UPDATE_PRODUCT_MUTATION,
  DELETE_PRODUCT_MUTATION,
} from '../../../graphql'
import { mockUser, mockProduct, mockImage, mockVariant, mockShopProductsVariables } from '../utils';


const productQueryMock = {
  request: { query: PRODUCT_QUERY, variables: { id: mockProduct.id } },
  result: {
    data: {
      product: {
        ...mockProduct,
        __typename: mockProduct.__typename,
        image: mockImage,
        user: {
          __typename: mockUser.__typename,
          id: mockUser.id,
          name: mockUser.name,
        },
        variants: [{
          ...mockVariant,
          product: {
            __typename: mockProduct.__typename,
            id: mockProduct.id,
            image: mockImage
          }
        }]
      },
    },
  },
};

const productQueryNoVariantMock = {
  request: { query: PRODUCT_QUERY, variables: { id: mockProduct.id } },
  result: {
    data: {
      product: {
        ...mockProduct,
        __typename: mockProduct.__typename,
        image: mockImage,
        user: {
          __typename: mockUser.__typename,
          id: mockUser.id,
          name: mockUser.name,
        },
        variants: []
      },
    },
  },
};

const createProductMutationMock = {
  request: {
    query: CREATE_PRODUCT_MUTATION,
    variables: {
      department: mockProduct.department,
      title: mockProduct.title,
      description: mockProduct.description,
      category: mockProduct.category,
      brand: mockProduct.brand,
      online: mockProduct.online,
      imageId: mockImage.id,
    },
  },
  result: {
    data: {
      createProduct: {
        __typename: 'Product',
        id: mockProduct.id,
      },
    },
  },
};

const createProductMutationErrorMock = {
  request: {
    query: CREATE_PRODUCT_MUTATION,
    variables: {
      department: 'tops',
      title: '',
      description: '',
      category: '',
      brand: '',
      online: false,
      imageId: mockImage.id,
    },
  },
  result: {
    errors: [{ message: 'ack!' }],
  }
};

const updateProductMutationMock = {
  request: {
    query: UPDATE_PRODUCT_MUTATION,
    variables: {
      id: mockProduct.id,
      department: mockProduct.department,
      title: mockProduct.title,
      description: mockProduct.description,
      category: mockProduct.category,
      brand: mockProduct.brand,
      online: mockProduct.online,
      imageId: mockImage.id,
    },
  },
  result: {
    data: {
      updateProduct: {
        __typename: 'Product',
        id: mockProduct.id,
      },
    },
  },
};

const updateProductMutationErrorMock = {
  request: {
    query: UPDATE_PRODUCT_MUTATION,
    variables: {
      id: mockProduct.id,
      department: 'tops',
      title: '',
      description: '',
      category: '',
      brand: '',
      online: false,
      imageId: mockImage.id,
    },
  },
  result: {
    errors: [{ message: 'ack!' }],
  }
};

const deleteProductMutationMock = {
  request: { query: DELETE_PRODUCT_MUTATION, variables: { id: mockProduct.id } },
  result: {
    data: {
      deleteProduct: {
        __typename: 'Product',
        id: mockProduct.id,
      },
    },
  },
};

const shopProductsQueryNameEmptyMock = {
  request: { query: SHOP_PRODUCTS_QUERY, variables: { name: mockUser.name }},
  result: {
    data: {
      products: [],
    },
  },
};

const shopProductsQueryProductMock = {
  request: { query: SHOP_PRODUCTS_QUERY, variables: { ...mockShopProductsVariables }},
  result: {
    data: {
      products: [{
        ...mockProduct,
        image: mockImage,
        user: {
          __typename: mockUser.__typename,
          id: mockUser.id,
          name: mockUser.name,
        },
        variants: [{
          ...mockVariant,
          product: {
            __typename: mockProduct.__typename,
            id: mockProduct.id,
            image: mockImage,
          }
        }]
      }],
    },
  },
};


export {
  productQueryMock, productQueryNoVariantMock,
  shopProductsQueryNameEmptyMock, shopProductsQueryProductMock,
  createProductMutationMock, createProductMutationErrorMock,
  updateProductMutationMock, updateProductMutationErrorMock,
  deleteProductMutationMock,
}
