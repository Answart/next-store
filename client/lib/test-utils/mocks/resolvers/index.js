export {
  localUserQueryEmptyCartMock, localUserQueryNoUserMock, localUserQueryCartItemMock,
} from './local';
export {
  userQueryEmptyCartMock, userQueryNoUserMock, userQueryCartItemMock,
  signupMutationMock,
  signinMutationMock,
  signoutMutationMock,
  requestPasswordResetMutationMock, requestPasswordResetMutationErrorMock,
  resetPasswordMutationMock, resetPasswordMutationErrorMock,
  updatePermissionsMutationMock,
} from './user';
export {
  createImageMutationMock, createImageMutationErrorMock,
} from './image';
export {
  productQueryMock, productQueryErrorMock, productQueryNoProductMock, productQueryNoVariantMock,
  shopProductsQueryFilterMock,
  createProductMutationMock, createProductMutationErrorMock,
  updateProductMutationMock, updateProductMutationErrorMock,
  deleteProductMutationMock,
} from './product';
export {
  createProductVariantMutationMock, createProductVariantMutationErrorMock,
  updateProductVariantMutationMock, updateProductVariantMutationErrorMock,
  deleteProductVariantMutationMock
} from './variant';
export {
  addToCartMutationMock,
  updateCartItemMutationMock,
  removeFromCartMutationMock
} from './cartItem';
