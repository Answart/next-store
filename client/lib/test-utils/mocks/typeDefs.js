import {
  fakeUser, fakeImage, fakeProduct, fakeVariant, fakeCartItem, fakeOrder, fakeOrderItem,
} from '../utils';


const mockUser = fakeUser();
const mockUser2 = fakeUser();
const mockImage = fakeImage();
const mockProduct = fakeProduct();
const mockVariant = fakeVariant();
const mockCartItem = fakeCartItem();
const mockOrder = fakeOrder();
const mockOrderItem = fakeOrderItem();

mockOrder.items.push(mockOrderItem);

const mockUsers = [
  mockUser,
  {
    ...mockUser2,
    id: '87654321',
    permissions: ['USER'],
  }
];
const mockProducts = [{
  ...mockProduct,
  image: mockImage,
  user: {
    __typename: mockUser.__typename,
    id: mockUser.id,
    name: mockUser.name,
  },
  variants: [{
    ...mockVariant,
    product: {
      __typename: mockProduct.__typename,
      id: mockProduct.id,
      image: mockImage,
    }
  }]
}];

const mockImageVariables = {
  cloudinary_id: mockImage.cloudinary_id,
  name: mockImage.name,
  height: mockImage.height,
  width: mockImage.width,
  transformation: mockImage.transformation,
  image_url: mockImage.image_url,
  large_image_url: mockImage.large_image_url
};

const mockShopProductsVariables = {
  name: mockUser.name,
  orderBy: 'createdAt_DESC',
  skip: 0,
  first: 1
 };


export {
  mockUser, mockUsers,
  mockImage, mockImageVariables,
  mockProduct, mockProducts, mockShopProductsVariables,
  mockVariant,
  mockCartItem,
  mockOrder, mockOrderItem,
};
