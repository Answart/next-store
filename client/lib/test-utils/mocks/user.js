import {
  CURRENT_USER_QUERY,
  SIGNOUT_MUTATION,
  REQUEST_PASSWORD_RESET_MUTATION,
} from '../../../graphql'
import { fakeUser } from '../utils';
import { mockCartItem } from './cart';

const mockUser = fakeUser();


const userQueryEmptyCartMock = {
  request: { query: CURRENT_USER_QUERY },
  result: {
    data: {
      me: {
        ...mockUser,
        cart: [],
      },
    },
  },
};

const userQueryNoUserMock = {
  request: { query: CURRENT_USER_QUERY },
  result: {
    data: { me: null }
  },
};

const userQueryCartItemMock = overrides => ({
  request: { query: CURRENT_USER_QUERY },
  result: {
    data: {
      me: {
        ...mockUser,
        cart: [{
          ...mockCartItem,
          ...overrides
        }],
      },
    },
  },
});

const signoutMutationMock = {
  request: { query: SIGNOUT_MUTATION },
  result: {
    data: {
      signout: {
        __typename: 'Message',
        success: true,
        message: 'Goodbye!'
      },
    },
  },
};

const requestPasswordResetMutationMock = {
  request: { query: REQUEST_PASSWORD_RESET_MUTATION, variables: { email: 'answart@sbcglobal.net' } },
  result: {
    data: {
      requestPasswordReset: {
        __typename: 'Message',
        success: true,
        message: 'Your request has been sent! Check your email.',
      }
    },
  },
};

const requestPasswordResetMutationErrorMock = {
  request: { query: REQUEST_PASSWORD_RESET_MUTATION, variables: { email: 'err@g.c' } },
  result: {
    data: {
      requestPasswordReset: {
        __typename: 'Message',
        success: false,
        message: 'No such user found for email err@g.c',
      }
    },
  },
};


export {
  mockUser,
  userQueryEmptyCartMock, userQueryNoUserMock, userQueryCartItemMock,
  signoutMutationMock,
  requestPasswordResetMutationMock, requestPasswordResetMutationErrorMock
};