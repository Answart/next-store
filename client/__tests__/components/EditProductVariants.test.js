import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import EditProductVariants from '../../components/EditProductVariants';
import { mockProduct, mockVariant } from '../../lib/test-utils/mocks';


describe('<EditProductVariants />', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(
      <MockedProvider mocks={[]}>
        <EditProductVariants product={{
          ...mockProduct,
          variants: [{ ...mockVariant }]
        }} />
      </MockedProvider>
    );
  });
  afterAll(() => {
    wrapper.unmount();
    jest.clearAllMocks();
  });

  it('matches the snap shot', async () => {
    await wait();
    wrapper.update();
    expect(toJSON(wrapper.find('EditProductVariants'))).toMatchSnapshot();
  });

  it('renders properly before selecting a variant', async () => {
    expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
      currentVariant: null,
      message: ''
    });
    expect(wrapper.find('DisplayMessage').length).toBe(1);
    expect(wrapper.find('DisplayMessage').props().success).toBe('');
    expect(wrapper.find('DisplayMessage').text()).toBe(null);
    const beforeChoose = wrapper.find('.edit-prdct-var-choose');
    expect(beforeChoose.length).toBe(1);
    expect(beforeChoose.find('.edit-prdct-lbl').text()).toBe('1. Choose Selection to Update');
    expect(beforeChoose.find('Product').length).toBe(1);
    expect(wrapper.find('.edit-prdct-var-form').length).toBe(0);
  });

  describe('2. Update Selection', () => {

    it('selecting a variant changes the instance state', async () => {
      expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
        currentVariant: null,
        message: ''
      });
      const mockEvent = { preventDefault: jest.fn() };
      wrapper.find('EditProductVariants').instance().selectVariant(mockEvent, mockVariant);
      await wait(50);
      wrapper.update();
      expect(mockEvent.preventDefault).toHaveBeenCalled();
      expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
        currentVariant: mockVariant,
        message: ''
      });
      jest.clearAllMocks();
    });

    it('renders properly after selecting a variant', async () => {
      expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
        currentVariant: mockVariant,
        message: ''
      });
      expect(wrapper.find('.edit-prdct-var-choose').length).toBe(0);
      const afterChoose = wrapper.find('.edit-prdct-var-form');
      expect(afterChoose.length).toBe(1);
      expect(afterChoose.find('.edit-prdct-lbl').text()).toBe('2. Update Selection');
      expect(afterChoose.find('UpdateProductVariantForm').length).toBe(1);
      expect(afterChoose.find('DeleteProductVariant').length).toBe(1);
    });

    it('UpdateProductVariantForm goBack fn calls properly', async () => {
      expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
        currentVariant: mockVariant,
        message: ''
      });
      wrapper.find('UpdateProductVariantForm').props().goBack();
      expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
        currentVariant: null,
        message: ''
      });
    });

    it('DeleteProductVariant postDelete fn calls properly', async () => {
      wrapper.find('EditProductVariants').instance().selectVariant(null, mockVariant);
      await wait(50);
      wrapper.update();
      expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
        currentVariant: mockVariant,
        message: ''
      });
      wrapper.find('DeleteProductVariant').props().postDelete();
      expect(wrapper.find('EditProductVariants').instance().state).toMatchObject({
        currentVariant: null,
        message: `Deleted selection with size '${mockVariant.size}' and color '${mockVariant.color}'.`
      });
    });
  });
});
