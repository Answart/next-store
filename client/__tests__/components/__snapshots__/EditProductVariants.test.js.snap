// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProductVariants /> matches the snap shot 1`] = `
<EditProductVariants
  product={
    Object {
      "__typename": "Product",
      "brand": "Peggs",
      "category": "jewelry",
      "department": "accessories",
      "description": "Limited Edition watch from the 2018 Fall fashion line.",
      "id": "pr0duct1d",
      "image": Object {
        "__typename": "Image",
        "cloudinary_id": "111111",
        "height": 22,
        "id": "1m4g31d",
        "image_url": "peggswatch1.jpg",
        "large_image_url": "peggswatch2.jpg",
        "name": "peggswatch1",
        "transformation": "",
        "width": 22,
      },
      "online": false,
      "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
      "user": Object {
        "__typename": "User",
        "cart": Array [],
        "email": "Waldo.Yundt@Mosciski.biz",
        "id": "us3r1d",
        "name": "Ms. Josue McCullough",
        "password": "unique-password",
        "permissions": Array [
          "ADMIN",
        ],
      },
      "variants": Array [
        Object {
          "__typename": "Variant",
          "availability": "4 in Stock!",
          "color": "white",
          "id": "v4r13nt1d",
          "image": Object {
            "__typename": "Image",
            "cloudinary_id": "111111",
            "height": 22,
            "id": "1m4g31d",
            "image_url": "peggswatch1.jpg",
            "large_image_url": "peggswatch2.jpg",
            "name": "peggswatch1",
            "transformation": "",
            "width": 22,
          },
          "price": 35,
          "product": Object {
            "__typename": "Product",
            "brand": "Peggs",
            "category": "jewelry",
            "department": "accessories",
            "description": "Limited Edition watch from the 2018 Fall fashion line.",
            "id": "pr0duct1d",
            "image": Object {
              "__typename": "Image",
              "cloudinary_id": "111111",
              "height": 22,
              "id": "1m4g31d",
              "image_url": "peggswatch1.jpg",
              "large_image_url": "peggswatch2.jpg",
              "name": "peggswatch1",
              "transformation": "",
              "width": 22,
            },
            "online": false,
            "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
            "user": Object {
              "__typename": "User",
              "cart": Array [],
              "email": "Princess.Ernser@Dibbert.net",
              "id": "us3r1d",
              "name": "Dr. Randy Gleason",
              "password": "unique-password",
              "permissions": Array [
                "ADMIN",
              ],
            },
            "variants": Array [],
          },
          "quantity": 4,
          "sale": true,
          "salePrice": 30,
          "size": "S",
        },
      ],
    }
  }
>
  <ProductStyles__StyledEditProductVariants>
    <div
      className="ProductStyles__StyledEditProductVariants-sc-2231u4-4 feaxHC"
    >
      <DisplayMessage
        error={Object {}}
        success=""
      />
      <div
        className="edit-prdct-var-choose"
      >
        <div
          className="edit-prdct-lbl"
        >
          1. Choose Selection to Update
        </div>
        <div
          className="edit-prdct-content"
        >
          <Product
            demoView={true}
            product={
              Object {
                "__typename": "Product",
                "brand": "Peggs",
                "category": "jewelry",
                "department": "accessories",
                "description": "Limited Edition watch from the 2018 Fall fashion line.",
                "id": "pr0duct1d",
                "image": Object {
                  "__typename": "Image",
                  "cloudinary_id": "111111",
                  "height": 22,
                  "id": "1m4g31d",
                  "image_url": "peggswatch1.jpg",
                  "large_image_url": "peggswatch2.jpg",
                  "name": "peggswatch1",
                  "transformation": "",
                  "width": 22,
                },
                "online": false,
                "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
                "user": Object {
                  "__typename": "User",
                  "cart": Array [],
                  "email": "Waldo.Yundt@Mosciski.biz",
                  "id": "us3r1d",
                  "name": "Ms. Josue McCullough",
                  "password": "unique-password",
                  "permissions": Array [
                    "ADMIN",
                  ],
                },
                "variants": Array [
                  Object {
                    "__typename": "Variant",
                    "availability": "4 in Stock!",
                    "color": "white",
                    "id": "v4r13nt1d",
                    "image": Object {
                      "__typename": "Image",
                      "cloudinary_id": "111111",
                      "height": 22,
                      "id": "1m4g31d",
                      "image_url": "peggswatch1.jpg",
                      "large_image_url": "peggswatch2.jpg",
                      "name": "peggswatch1",
                      "transformation": "",
                      "width": 22,
                    },
                    "price": 35,
                    "product": Object {
                      "__typename": "Product",
                      "brand": "Peggs",
                      "category": "jewelry",
                      "department": "accessories",
                      "description": "Limited Edition watch from the 2018 Fall fashion line.",
                      "id": "pr0duct1d",
                      "image": Object {
                        "__typename": "Image",
                        "cloudinary_id": "111111",
                        "height": 22,
                        "id": "1m4g31d",
                        "image_url": "peggswatch1.jpg",
                        "large_image_url": "peggswatch2.jpg",
                        "name": "peggswatch1",
                        "transformation": "",
                        "width": 22,
                      },
                      "online": false,
                      "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
                      "user": Object {
                        "__typename": "User",
                        "cart": Array [],
                        "email": "Princess.Ernser@Dibbert.net",
                        "id": "us3r1d",
                        "name": "Dr. Randy Gleason",
                        "password": "unique-password",
                        "permissions": Array [
                          "ADMIN",
                        ],
                      },
                      "variants": Array [],
                    },
                    "quantity": 4,
                    "sale": true,
                    "salePrice": 30,
                    "size": "S",
                  },
                ],
              }
            }
            variantAction={[Function]}
            variantActionLabel="Select"
          >
            <ProductStyles__StyledProduct>
              <div
                className="ProductStyles__StyledProduct-sc-2231u4-0 dOvnhj"
              >
                <div
                  className="prdct-imgs"
                >
                  <img
                    alt="Peggs Gold Edition Analog Watch Peggs Gold Edition Analog "
                    height="640"
                    src="peggswatch2.jpg"
                    width="450"
                  />
                </div>
                <div
                  className="prdct-content"
                >
                  <div
                    className="prdct-padding"
                    id="prdct-title"
                  >
                    <Link
                      href={
                        Object {
                          "pathname": "/buy",
                          "query": Object {
                            "id": "pr0duct1d",
                          },
                        }
                      }
                    >
                      <a
                        className="prdct-title"
                        href="/buy?id=pr0duct1d"
                        onClick={[Function]}
                      >
                        Peggs Gold Edition Analog Watch Peggs Gold Edition Analog 
                      </a>
                    </Link>
                  </div>
                  <ByCreator
                    name="Ms. Josue McCullough"
                    online={true}
                  >
                    <div
                      className="prdct-creator prdct-padding"
                    >
                      By
                      <Link
                        href={
                          Object {
                            "pathname": "/shop",
                            "query": Object {
                              "name": "Ms. Josue McCullough",
                              "online": true,
                            },
                          }
                        }
                      >
                        <a
                          className="undrln-btn"
                          href="/shop?name=Ms.%20Josue%20McCullough&online=true"
                          onClick={[Function]}
                        >
                          Ms. Josue McCullough
                        </a>
                      </Link>
                    </div>
                  </ByCreator>
                  <div
                    className="prdct-padding"
                    id="prdct-online"
                  >
                    <i>
                      Offline
                    </i>
                  </div>
                  <ProductVariants
                    demoView={true}
                    online={false}
                    variantAction={[Function]}
                    variantActionLabel="Select"
                    variants={
                      Array [
                        Object {
                          "__typename": "Variant",
                          "availability": "4 in Stock!",
                          "color": "white",
                          "id": "v4r13nt1d",
                          "image": Object {
                            "__typename": "Image",
                            "cloudinary_id": "111111",
                            "height": 22,
                            "id": "1m4g31d",
                            "image_url": "peggswatch1.jpg",
                            "large_image_url": "peggswatch2.jpg",
                            "name": "peggswatch1",
                            "transformation": "",
                            "width": 22,
                          },
                          "price": 35,
                          "product": Object {
                            "__typename": "Product",
                            "brand": "Peggs",
                            "category": "jewelry",
                            "department": "accessories",
                            "description": "Limited Edition watch from the 2018 Fall fashion line.",
                            "id": "pr0duct1d",
                            "image": Object {
                              "__typename": "Image",
                              "cloudinary_id": "111111",
                              "height": 22,
                              "id": "1m4g31d",
                              "image_url": "peggswatch1.jpg",
                              "large_image_url": "peggswatch2.jpg",
                              "name": "peggswatch1",
                              "transformation": "",
                              "width": 22,
                            },
                            "online": false,
                            "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
                            "user": Object {
                              "__typename": "User",
                              "cart": Array [],
                              "email": "Princess.Ernser@Dibbert.net",
                              "id": "us3r1d",
                              "name": "Dr. Randy Gleason",
                              "password": "unique-password",
                              "permissions": Array [
                                "ADMIN",
                              ],
                            },
                            "variants": Array [],
                          },
                          "quantity": 4,
                          "sale": true,
                          "salePrice": 30,
                          "size": "S",
                        },
                      ]
                    }
                  >
                    <ProductStyles__StyledProductVariants>
                      <div
                        className="ProductStyles__StyledProductVariants-sc-2231u4-1 ndRtn"
                      >
                        <div
                          className="prdct-padding"
                        >
                          <PriceTag
                            price={35}
                            sale={true}
                            salePrice={30}
                          >
                            <PriceTag__StyledPriceTag>
                              <div
                                className="PriceTag__StyledPriceTag-sc-119rsfa-0 ePOheW"
                              >
                                <span
                                  className="line-through"
                                >
                                  $35
                                </span>
                                <span
                                  className="price-tag-padding price-tag-priceSale"
                                >
                                  $30
                                </span>
                              </div>
                            </PriceTag__StyledPriceTag>
                          </PriceTag>
                        </div>
                        <div
                          className="prdct-padding"
                          id="prod-var-sizes"
                        >
                          <div
                            className="prdct-padding filter-sizes"
                          >
                            <button
                              className="undrln-btn sample-selected"
                              id="prod-var-size-S"
                              key="0"
                              name="size"
                              onClick={[Function]}
                              title="Select size: S"
                              value="S"
                            >
                              S
                            </button>
                          </div>
                        </div>
                        <div
                          className="prdct-padding"
                          id="prod-var-colors"
                        >
                          <div
                            id="prod-var-colors-title"
                          >
                            White
                          </div>
                          <div
                            className="prdct-padding filter-colors"
                          >
                            <button
                              className="sample-selected"
                              id="prod-var-color-white"
                              key="0"
                              name="color"
                              onClick={[Function]}
                              title="Select color: White"
                              value="white"
                            >
                              <div
                                className="color-sphere-sample white-color-sample"
                              />
                            </button>
                          </div>
                        </div>
                        <div
                          className="prdct-padding"
                          id="prod-var-avail"
                        >
                          <i>
                            4 in Stock!
                          </i>
                        </div>
                        <div
                          className="prdct-padding"
                          id="prod-var-action"
                        >
                          <button
                            className="big-btn"
                            disabled={false}
                            id="prod-var-action-btn"
                            onClick={[Function]}
                          >
                            Select
                          </button>
                        </div>
                      </div>
                    </ProductStyles__StyledProductVariants>
                  </ProductVariants>
                  <div
                    className="prdct-desc prdct-padding"
                    id="prdct-desc"
                  >
                    <div
                      id="prdct-description"
                    >
                      <strong>
                        Description:
                      </strong>
                      <p>
                        Limited Edition watch from the 2018 Fall fashion line.
                      </p>
                    </div>
                    <div
                      id="prdct-brand"
                    >
                      <strong>
                        Brand:
                      </strong>
                      <p>
                        Peggs
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </ProductStyles__StyledProduct>
          </Product>
        </div>
      </div>
    </div>
  </ProductStyles__StyledEditProductVariants>
</EditProductVariants>
`;
